application/src/main/java/org/togetherjava/tjbot/features/jshell/aws/JShellAWSCommand.java

```diff
@@ -0,0 +1,198 @@
+package org.togetherjava.tjbot.features.jshell.aws;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.SlashCommandAdapter;
+import org.togetherjava.tjbot.features.jshell.aws.exceptions.JShellAPIException;
+
+import java.awt.Color;
+
+/**
+ * This class contains the complete logic for the /jshell-aws command.
+ *
+ * @author Suraj Kumar
+ */
+public class JShellAWSCommand extends SlashCommandAdapter {
+    private static final Logger logger = LogManager.getLogger(JShellAWSCommand.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final String CODE_PARAMETER = "code";
+    private final JShellService jShellService;
+
+    /**
+     * Constructs a new JShellAWSCommand
+     *
+     * @param jShellService The service class to make requests against AWS
+     */
+    public JShellAWSCommand(JShellService jShellService) {
+        super("jshell-aws", "Execute Java code in Discord!", CommandVisibility.GUILD);
+        getData().addOption(OptionType.STRING, CODE_PARAMETER, "The code to execute using JShell");
+        this.jShellService = jShellService;
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        Member member = event.getMember();
+
+        if (member == null) {
+            event.reply("Member that executed the command is no longer available, won't execute")
+                .queue();
+            return;
+        }
+
+        logger.info("JShell AWS invoked by {} in channel {}", member.getAsMention(),
+                event.getChannelId());
+
+        OptionMapping input = event.getOption(CODE_PARAMETER);
+
+        if (input == null || input.getAsString().isEmpty()) {
+            EmbedBuilder eb = new EmbedBuilder();
+            eb.setDescription(member.getAsMention()
+                    + ", you forgot to provide the code for JShell to evaluate or it was too short!\nTry running the command again and make sure to select the code option");
+            eb.setColor(Color.ORANGE);
+            event.replyEmbeds(eb.build()).queue();
+            return;
+        }
```

Comment:
Check, but I think true is the default value

Reply:
Added it anyway, it enforces readability ig

Reply:
I double checked, required is false by default
<img width="841" alt="Screenshot 2024-09-14 at 4 37 42 PM" src="https://github.com/user-attachments/assets/72bfa037-56fe-49f9-8c80-156d648bc0fa">

Since you made _code_ param as mandatory, you no longer need to check for input being null or empty string